src/H5Ocache.c

```diff
@@ -1268,6 +1268,11 @@ H5O__chunk_deserialize(H5O_t *oh, haddr_t addr, size_t chunk_size, const uint8_t
         UINT16DECODE(chunk_image, mesg_size);
         if (mesg_size != H5O_ALIGN_OH(oh, mesg_size))
             HGOTO_ERROR(H5E_OHDR, H5E_CANTLOAD, FAIL, "message not aligned");
+        if (mesg_size > (size_t)(p_end - chunk_image))
```

Comment:
It may make more sense to use the `H5_IS_BUFFER_OVERFLOW` macro here, as in:

```C
if (H5_IS_BUFFER_OVERFLOW(chunk_image, mesg_size, p_end))
    error;
```

as it does the same thing and accounts for the fact that you need `p_end - chunk_image + 1`.